src/relay/op/nn/nn.cc

```diff
@@ -143,5 +143,162 @@ RELAY_REGISTER_UNARY_OP("relay.op.nn._make.", "relu")
 .set_support_level(1)
 .add_type_rel("Identity", IdentityRel);
 
+// Dropout
+TVM_REGISTER_NODE_TYPE(DropoutAttrs);
+
+Expr MakeDropout(Expr data, double rate) {
+  auto attrs = make_node<DropoutAttrs>();
+  attrs->rate = rate;
+  static const Op& op = Op::Get("nn.dropout");
+  return CallNode::make(op, {data}, Attrs(attrs), {});
+}
+
+TVM_REGISTER_API("relay.op.nn._make.dropout")
+.set_body([](const TVMArgs& args, TVMRetValue* rv) {
+    runtime::detail::unpack_call<Expr, 2>(MakeDropout, args, rv);
+  });
+
+RELAY_REGISTER_OP("nn.dropout")
+.describe(R"code(Applies the dropout operation to the input array.
+
+During training, each element of the input is set to zero with probability ``p``.
+The whole array is rescaled by ``1/(1-p)`` to keep the expected sum of the input unchanged.
+
+)code" TVM_ADD_FILELINE)
+.set_num_inputs(1)
+.add_argument("data", "Tensor", "Input to which dropout will be applied.")
+.set_support_level(1)
+.add_type_rel("Identity", IdentityRel);
```

Comment:
Dropout has 2 outputs,  I think you have to add as TupleTypeNode

Reply:
What is the second output? The NNVM doc for dropout only makes reference to a single result tensor, but perhaps I'm missing something.

Reply:
https://github.com/dmlc/tvm/blob/4e309e67a6c23411eec941000e469bd4e17cef92/nnvm/src/top/nn/nn.cc#L167

https://github.com/dmlc/tvm/blob/4e309e67a6c23411eec941000e469bd4e17cef92/nnvm/src/top/nn/nn.cc#L175

Here I can see 2 outputs, output and mask.

Reply:
Ah, you're right. The registry also says that only one of the outputs is visible, so presumably the "mask" output isn't exposed. I see no reference to the mask output anywhere else, so what characteristics should it have?

Reply:
Ah well I can infer based on `less` and `greater` that the mask tensor should have the same shape. I'll  put it in and see if other reviewers think this output should be exposed.